numpy/core/src/umath/absolute.dispatch.cpp

```diff
@@ -0,0 +1,129 @@
+/*@targets
+ * $maxopt baseline
+ * avx512_skx
+ * asimd
+ */
+
+#define _UMATHMODULE
+#define _MULTIARRAYMODULE
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#include <hwy/highway.h>
+#include "hwy/aligned_allocator.h"
+
+#include "numpy/ndarraytypes.h"
+#include "numpy/npy_common.h"
+#include "numpy/npy_math.h"
+#include "numpy/utils.h"
+
+#include "loops_utils.h"
+#include "fast_loop_macros.h"
+
+namespace numpy {
+namespace HWY_NAMESPACE {  // required: unique per target
+
+// Can skip hn:: prefixes if already inside hwy::HWY_NAMESPACE.
+namespace hn = hwy::HWY_NAMESPACE;
+
+// Alternative to per-function HWY_ATTR: see HWY_BEFORE_NAMESPACE
+template <typename T>
+HWY_ATTR void SuperAbsolute(char **args, npy_intp const *dimensions, npy_intp const *steps) {
+  const T* HWY_RESTRICT input_array = (const T*) args[0];
+  T* HWY_RESTRICT output_array = (T*) args[1];
+  const size_t size = dimensions[0];
+  const hn::ScalableTag<T> d;
+  
+    if (is_mem_overlap(input_array, steps[0], output_array, steps[1], size)) {
+      for (size_t i = 0; i < size; i++) {
+        const auto in = hn::LoadN(d, input_array + i, 1);
+        auto x = hn::Abs(in);
+        hn::StoreN(x, d, output_array + i, 1);
+      }
+    } else if (IS_UNARY_CONT(input_array, output_array)) {
+      size_t full = size & -hn::Lanes(d);
+      size_t remainder = size - full;
+      for (size_t i = 0; i < full; i += hn::Lanes(d)) {
+        const auto in = hn::LoadU(d, input_array + i);
+        auto x = hn::Abs(in);
+        hn::StoreU(x, d, output_array + i);
+      }
+      if (remainder) {
+        const auto in = hn::LoadN(d, input_array + full, remainder);
+        auto x = hn::Abs(in);
+        hn::StoreN(x, d, output_array + full, remainder);
+      }
+    } else {
+      using TI = hwy::MakeSigned<T>;
+      const hn::Rebind<TI, hn::ScalableTag<T>> di;
+
+      const int lsize = sizeof(input_array[0]);
+      const npy_intp ssrc = steps[0] / lsize;
+      const npy_intp sdst = steps[1] / lsize;
+      auto load_index = hwy::AllocateAligned<TI>(hn::Lanes(d));
+      for (size_t i = 0; i < hn::Lanes(d); ++i) {
```

Comment:
We could actually vectorize this, just Mul() for <= 32 bit types, and perhaps converting to double and then Mul of that for 64-bit?

Reply:
Here's an example of how to specialize a generic function for certain types:
https://github.com/libjxl/libjxl/blob/e6202f7181eff36c78bfdb79aa9bd45c3d1d614b/lib/jxl/rational_polynomial-inl.h#L29